### PR TITLE
Student: submit responses: allow student to add a comment for each response (MCQ questions) #7239

### LINKED ISSUE #7239: Student: submit responses: allow student to add a comment for each response
When students submit responses, it is useful if the system allows student to add an explanatory comment. It also avoid the need to add questions such as 'explain your response to the previous question' to collect justifications.

This might have to be done as multiple PRs, one per question type. We can start with MCQ questions.

### PR DESCRIPTION
**Pull request by Mynk96**
Fixes #7239

### CODE REVIEW

**Comment by Mynk96** [comment]
@chowyb this PR contains updated way of testing comments feature. Please have a look. Also, should I revert changes merged in the branch `add-student-response-comment` from PR #7817 ?

**Comment by Mynk96** [comment]
@chowyb Please review!

**Comment by chowyb** [comment]
@Mynk96 changes here seem fine for the most part, although again the dependency has to be made obvious (eg. it's not immediately clear to someone looking at the code as to why the method testing for deletion is checking for successful edits at the start of the method)

For the changes to tests, I had originally hoped you would handle it in the currently open PR (#7955) instead of opening a new PR for it, but since this PR is already open, it should be possible to just merge this into `add-student-response-comment` directly once this PR is done (since most of the earlier commits in this PR are already inside `add-student-response-comment` in the squashed commit) without having to revert #7817.  This would also mean you'll have to update #7955 with the updated `add-student-response-comment` once this PR is merged in.

**Comment by Mynk96** [comment]
@chowyb Should I change method name as suggested in PR #7955 or should I change test case description to show dependency?

**Comment by Mynk96** [comment]
@chowyb I have updated tests descriptions please Review!

**Comment by chowyb** [comment]
@Mynk96 changing the description seems okay, but I'd suggest changing the method name as well, as developers should not have to look inside the method itself to discover that there is a dependency. Applies to this and #7955.

**Comment by Mynk96** [comment]
@chowyb Please Review!

File: src/test/java/teammates/test/cases/browsertests/InstructorEditStudentFeedbackPageUiTest.java [hunk 1]
```diff
@@ -55,7 +76,7 @@ private void testEditResponse() throws Exception {
                                                   "student1InIESFPTCourse@gmail.tmt",
                                                   "student1InIESFPTCourse@gmail.tmt");
 
-        assertEquals("Student 1 self feedback.", fr.getResponseDetails().getAnswerString());
+        assertEquals("<p>Student 1 self feedback.</p>", fr.getResponseDetails().getAnswerString());
```
Comment (chowyb) [inline src/test/java/teammates/test/cases/browsertests/InstructorEditStudentFeedbackPageUiTest.java]: Why was this changed?

Comment (Mynk96) [inline src/test/java/teammates/test/cases/browsertests/InstructorEditStudentFeedbackPageUiTest.java]: It gave an assertion error on running test. Checking the other asserts in the file I don't know why this assert didn't have a `<p>/<p>` tag. So, I had to change to get green on tests

**Comment by chowyb** [comment]
Seems like there are now a few merge conflicts that need to be resolved (which I hadn't expected to happen). You'll likely have to update the branch by merging from `add-student-response-comment`.

**Comment by Mynk96** [comment]
@chowyb resolved merge conflicts!

**Comment by chowyb** [comment]
@Mynk96 PR is now merged, and as such I'd expect merge conflicts to happen on #7955 too, which you might want to fix in advance.
